frontend/src/pages/ActivityReport/Pages/components/Objective.js

```diff
@@ -170,14 +169,10 @@ export default function Objective({
   };
 
   const onUploadFile = async (files, _objective, setError) => {
-    // we save draft one of two ways, depending on whether it is a
-    // recipient report or not
-    await onSaveDraft();
-
-    // we also need to access the updated form data to
+    // we need to access the updated form data to
     // get the correct objective ids to attach to our API post
     const objectivesField = getValues(fieldArrayName);
-    const objectiveToAttach = objectivesField[index];
+    const objectiveToAttach = objectivesField.find((o) => o.id === objective.id);
```

Comment:
I wonder if we should be using 'selectedObjective' to get the ID here. In my previous ticket I found that when switching from an existing objective to a new objective. The 'objective' prop was not reliable.

Reply:
@AdamAdHocTeam Ok, I updated it to use selectedObjective